tests/test_cpp_select1.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017-2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include "utils/test_setup.h"
+#include "utils/TestSetup.hpp"
+
+void test_select1_cpp(void **unused) {
+    Snowflake::Client::Connection *cnx = TestSetup::connectionFactory();
+    SF_STATUS status;
+
+    try {
+        cnx->connect();
+        Snowflake::Client::Statement *stmt = new Snowflake::Client::Statement(*cnx);
+        std::string command("select 1;");
+        stmt->query(command);
+
+        int counter = 0;
+        while ((status = stmt->fetch()) == SF_STATUS_SUCCESS) {
+            assert_int_equal(stmt->column(1).asInt32(), 1);
+            ++counter;
+        }
```

Comment:
This looks much cleaner now.
So asInt32() method will defer converting the value to here instead of fetching and converting in fetch method, right?